be/src/exec/pipeline/pipeline_builder.cpp

```diff
@@ -383,7 +384,13 @@ OpFactories PipelineBuilderContext::maybe_gather_pipelines_to_one(RuntimeState*
         local_exchange_source->group_leader()->set_adaptive_blocking_event(std::move(merged_blocking_events));
     }
 
-    auto exchanger = std::make_shared<PassthroughExchanger>(mem_mgr, local_exchange_source.get());
+    std::shared_ptr<LocalExchanger> exchanger;
+    if (pass_through_type == LocalExchanger::PassThroughType::RANDOM) {
+        exchanger = std::make_shared<PassthroughExchanger>(mem_mgr, local_exchange_source.get());
+    } else {
+        exchanger = std::make_shared<DirectThroughExchanger>(mem_mgr, local_exchange_source.get());
```

Comment:
gather to one seems PassthroughExchanger has the same behaviour as DirectThroughExchanger